packages/react-core/src/demos/examples/AlertGroup/AlertGroupDemo.tsx

```diff
@@ -0,0 +1,423 @@
+  const maxDisplayedAlerts = 3;
+  const minAlerts = 0;
+  const maxAlerts = 100;
+  const alertTimeout = 7000;
+
+  const [isDrawerExpanded, setDrawerExpanded] = React.useState(false);
+  const [openDropdownKey, setOpenDropdownKey] = React.useState<React.Key | null>(null);
+  const [overflowMessage, setOverflowMessage] = React.useState<string>('');
+  const [maxDisplayed, setMaxDisplayed] = React.useState(maxDisplayedAlerts);
+  const [alerts, setAlerts] = React.useState<React.ReactElement<AlertProps>[]>([]);
+  const [notifications, setNotifications] = React.useState<NotificationProps[]>([]);
+
+  useEffect(() => {
+    setOverflowMessage(buildOverflowMessage());
+  }, [maxDisplayed, notifications, alerts]);
+
+  const handleNewNotification = (
+    title: string,
+    srTitle: string,
+    variant: NotificationProps['variant'],
+    key: React.Key,
+    timestamp: string,
+    description: string
+  ) => {
+    setNotifications(prevNotifications => [
+      { title, srTitle, variant, key, timestamp, description, isNotificationRead: false },
+      ...prevNotifications
+    ]);
+
+    if (!isDrawerExpanded) {
+      setAlerts(prevAlerts => [
+        <Alert
+          variant={variant}
+          title={title}
+          timeout={alertTimeout}
+          onTimeout={() => removeAlert(key)}
+          isLiveRegion
+          actionClose={
+            <AlertActionCloseButton title={title} variantLabel={`${variant} alert`} onClose={() => removeAlert(key)} />
+          }
+          key={key}
+          id={key.toString()}
+        >
+          <p>{description}</p>
+        </Alert>,
+        ...prevAlerts
+      ]);
+    }
+  };
+
+  const removeNotification = (key: React.Key) => {
+    setNotifications(prevNotifications => prevNotifications.filter(notification => notification.key !== key));
+  };
+
+  const removeAllNotifications = () => {
+    setNotifications([]);
+  };
+
+  const isNotificationRead = (key: React.Key) =>
+    notifications.find(notification => notification.key === key)?.isNotificationRead === true;
+
+  const markNotificationRead = (key: React.Key) => {
+    setNotifications(prevNotifications =>
+      prevNotifications.map(notification =>
+        notification.key === key ? { ...notification, isNotificationRead: true } : notification
+      )
+    );
+  };
+
+  const markAllNotificationsRead = () => {
+    setNotifications(prevNotifications =>
+      prevNotifications.map(notification => ({ ...notification, isNotificationRead: true }))
+    );
+  };
+
+  const getUnreadNotificationsNumber = () =>
+    notifications.filter(notification => notification.isNotificationRead === false).length;
+
+  const onNotificationBadgeClick = () => {
+    removeAllAlerts();
+    setDrawerExpanded(!isDrawerExpanded);
+  };
+
+  const onDropdownToggle = (id: React.Key, isActive: boolean) => {
+    if (!isActive) {
+      setOpenDropdownKey(null);
+      return;
+    }
+    setOpenDropdownKey(id);
```

Comment:
Nit: I would swap these and inverse the conditional as I find it's generally a little bit more straightforward for conditionals to not be inverted when it can be avoided.
```suggestion
    if (isActive) {
      setOpenDropdownKey(id);
      return;
    }
    setOpenDropdownKey(null);
```